src/server/scripts/EasternKingdoms/ZulGurub/boss_thekal.cpp

```diff
@@ -85,56 +98,147 @@ class boss_thekal : public CreatureScript
 
         struct boss_thekalAI : public BossAI
         {
-            boss_thekalAI(Creature* creature) : BossAI(creature, DATA_THEKAL)
+            boss_thekalAI(Creature* creature) : BossAI(creature, DATA_THEKAL), _instance(creature->GetInstanceScript())
```

Comment:
BossAI already stores instance script in "instance"variable https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/AI/ScriptedAI/ScriptedCreature.cpp#L450

Reply:
Right, I think I realized that halfway through the script (using instance everywhere else) but forgot to remove this :P Thanks!